pynest/examples/eprop_plasticity/eprop_supervised_classification_neuromorphic_mnist.py

```diff
@@ -433,42 +426,41 @@ def calculate_glorot_dist(fan_in, fan_out):
 # representative samples are used throughout the training process.
 
 
-def download_and_extract_dataset(url, dataset_directory="468j46mzdv-1"):
-    path = os.path.join(".", dataset_directory)
+def unzip(zip_file_path, extraction_path):
+    print(f"Extracting {zip_file_path}.")
+    with zipfile.ZipFile(zip_file_path, "r") as zip_file:
+        zip_file.extractall(extraction_path)
+    os.remove(zip_file_path)
 
-    expected_contents = ["Test", "Train"]
-    if os.path.exists(path) and all(os.path.exists(os.path.join(path, content)) for content in expected_contents):
-        print(f"\nThe directory '{path}' already exists with expected contents. Skipping download and extraction.")
-        return path
 
-    local_zip_filename = "dataset.zip"
+def download_and_extract_nmnist_dataset(save_path="./"):
+    nmnist_dataset = {
+        "url": "https://prod-dcd-datasets-cache-zipfiles.s3.eu-west-1.amazonaws.com/468j46mzdv-1.zip",
```

Comment:
This encapsulation is a very good idea since this "url" should not have been a parameter.